plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/SwitchDataset/index.tsx

```diff
@@ -15,6 +15,7 @@ const SwitchDataset: React.FC<BaseFieldProps<"switchDataset">> = ({
   value,
   editMode,
   configData,
+  getSelectedDataset,
```

Comment:
You don't need this. You can get it from dataset.selectedDataset

Reply:
The problem is that when you load a dataset that contains a switch dataset it does not download until you change the dataset.
So, I added a `useEffect` in the `SwitchDataset` component. Please check and let me know what you think.

Reply:
Yes, that is why I added this conditional statement.
```ts
    if (dataset.url) {
      const name = getNameFromPath(dataset.name);
      const format = normalizeExtension(dataset.format);
      const filename = createFileName(name, format);
      downloadData(dataset.url, filename);
    } else if (dataset.config?.data && dataset.selectedDataset) {
      const name = getNameFromPath(dataset.selectedDataset.name);
      const format = normalizeExtension(dataset.format);
      const filename = createFileName(name, format);
      downloadData(dataset.selectedDataset.url, filename);
    }
```

Reply:
Ok. Anyways we don't want field components doing any updates on initialization, so if you can remove the useEffect from the switchDataset.
Instead it would be better to add the selectedDataset when we add the dataset to the scene.

Reply:
I am sorry, I am not sure how to do this. can you give me more details please.

Reply:
What I was thinking was in the `handleProjectDatasetAdd` callback. Here we do some processing, checking components, add a defaultTemplate if there is one, attach a default selected group, and build overrides.
Here you can do the exact same thing we are doing with selectedGroup with selectedDataset!

Reply:
Done!